### PR TITLE
Call citizen apis and added unit test cases

### PR DESCRIPTION
**Pull request by DommarajuUmamaheswari**

### CODE REVIEW

File: src/Kmd.Momentum.Mea.Api/Citizen/CitizenController.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        ///</summary>
+        ///<response code="200">The data is loaded successfully</response>
+        ///<response code="400">Bad request</response>
+        ///<response code="404">Document is not found</response>
+        [HttpGet]
+        [ProducesResponseType(200)]
+        [ProducesResponseType(400)]
+        [ProducesResponseType(404)]
+        [Route("/citizenByCpr/{cpr}")]
+        [SwaggerOperation(OperationId = "GetCitizenByCpr")]
```
Comment (amitxagarwal) [inline src/Kmd.Momentum.Mea.Api/Citizen/CitizenController.cs]: why the casing here is different from the next operation

File: src/Kmd.Momentum.Mea.Api/Citizen/CitizenDataResponse.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
+﻿namespace Kmd.Momentum.Mea.Api
+{
+    public class CitizenDataResponse
+    {
+        public string Id { get; }
```
Comment (amitxagarwal) [inline src/Kmd.Momentum.Mea.Api/Citizen/CitizenDataResponse.cs]: check with the latest yml file if these are the latest fields we have to include

File: src/Kmd.Momentum.Mea.Api/Citizen/CitizenService.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+            if (response.StatusCode == System.Net.HttpStatusCode.OK)
+            {
+                var json = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+                return JsonConvert.DeserializeObject<string[]>(json);
+            }
+            throw new Exception(response.StatusCode.ToString());
+        }
+
+        public async Task<CitizenDataResponse> getCitizenByCpr(string cpr)
```
Comment (amitxagarwal) [inline src/Kmd.Momentum.Mea.Api/Citizen/CitizenService.cs]: All the methods names should be in Camel Case

File: src/Kmd.Momentum.Mea.Api/Citizen/CitizenService.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        private readonly IConfiguration _config;
+
+        public CitizenService(IHelperHttpClient citizenHttpClient, IConfiguration config)
+        {
+            _citizenHttpClient = citizenHttpClient;
+            _config = config;
+        }
+        public async Task<string[]> GetAllActiveCitizens()
+        {
+            HttpResponseMessage response = await _citizenHttpClient.GetMcaData(_config, new Uri($"{_config["McaUri"]}citizens/withActiveClassification"), "get").ConfigureAwait(false);
```
Comment (amitxagarwal) [inline src/Kmd.Momentum.Mea.Api/Citizen/CitizenService.cs]: method name should be like GetDataFromMomentumCore

File: src/Kmd.Momentum.Mea.Api/Citizen/EnrollmentStatus.cs [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Kmd.Momentum.Mea.Api
+{
+    public class EnrollmentStatus
+    {
+        public DateTimeOffset? EndDate { get; set; }
+        public int Status { get; set; }
```
Comment (amitxagarwal) [inline src/Kmd.Momentum.Mea.Api/Citizen/EnrollmentStatus.cs]: is this int or bool?

File: src/Kmd.Momentum.Mea.Api/Citizen/ICitizenService.cs [hunk 6]
```diff
@@ -0,0 +1,13 @@
+﻿using System.Threading.Tasks;
+
+namespace Kmd.Momentum.Mea.Api.Citizen
+{
+    public interface ICitizenService
+    {
+        Task<string[]> GetAllActiveCitizens();
+
+        Task<CitizenDataResponse> getCitizenByCpr(string cpr);
```
Comment (amitxagarwal) [inline src/Kmd.Momentum.Mea.Api/Citizen/ICitizenService.cs]: the casing of method names are not right

File: src/Kmd.Momentum.Mea.Api/Common/HelperHttpClient.cs [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Newtonsoft.Json.Linq;
+using System.Globalization;
+using Microsoft.Extensions.Configuration;
+
+namespace Kmd.Momentum.Mea.Api.Common
+{
+#pragma warning disable CA1001 // Types that own disposable fields should be disposable
```
Comment (amitxagarwal) [inline src/Kmd.Momentum.Mea.Api/Common/HelperHttpClient.cs]: make sure to fix this in the next PR. please create a work item to track this.

File: src/Kmd.Momentum.Mea.Api/Common/IHelperHttpClient.cs [hunk 8]
```diff
@@ -0,0 +1,15 @@
+﻿using Microsoft.Extensions.Configuration;
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Kmd.Momentum.Mea.Api.Common
+{
+    public interface IHelperHttpClient
+    {
+        Task<HttpResponseMessage> ReturnAuthorizationToken(IConfiguration config);
```
Comment (amitxagarwal) [inline src/Kmd.Momentum.Mea.Api/Common/IHelperHttpClient.cs]: are we calling this method from outside of httpclienthelper? if not then why this has to be a part of contract?

File: src/Kmd.Momentum.Mea.Api/Citizen/CitizenController.cs [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+using System.Threading.Tasks;
+
+namespace Kmd.Momentum.Mea.Api.Citizen
+{
+    [ApiController]
+    [Route("citizen")]
+    [Produces("application/json", "text/json")]
+    public class CitizenController : ControllerBase
+    {
+        private readonly ICitizenService _citizenService;
```
Comment (Ashish0410verma) [inline src/Kmd.Momentum.Mea.Api/Citizen/CitizenController.cs]: space ?

File: src/Kmd.Momentum.Mea.Api/Citizen/CitizenController.cs [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+        public CitizenController(ICitizenService citizenService)
+        {
+            _citizenService = citizenService ?? throw new ArgumentNullException(nameof(citizenService));            
+        }
+
+        ///<summary>
+        ///Get all active citizens
+        ///</summary>
+        ///<response code="200">The data is loaded successfully</response>
```
Comment (Ashish0410verma) [inline src/Kmd.Momentum.Mea.Api/Citizen/CitizenController.cs]: please change the XML comments messages on all controller methods

File: src/Kmd.Momentum.Mea.Api/Common/HelperHttpClient.cs [hunk 11]
```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Newtonsoft.Json.Linq;
+using System.Globalization;
```
Comment (Ashish0410verma) [inline src/Kmd.Momentum.Mea.Api/Common/HelperHttpClient.cs]: remove unused using blocks

File: src/Kmd.Momentum.Mea.Api/Citizen/CitizenController.cs (hunk 9 above)
Comment (Ashish0410verma) [reply src/Kmd.Momentum.Mea.Api/Citizen/CitizenController.cs]: fixed

**Comment by Ashish0410verma** [comment]
we need to be sure that our Api url path is same what has been given in yaml ?

**Comment by Ashish0410verma** [comment]
> we need to be sure that our Api url path is same what has been given in yaml ?

**Comment by DommarajuUmamaheswari** [comment]
> we need to be sure that our Api url path is same what has been given in yaml ?

> closing this comment as we have changed the api path as per the yaml

**Pull request merged**